Sources/Linq2DynamoDb.DataContext/IEntityWrapper.cs

```diff
@@ -7,5 +7,6 @@ public interface IEntityWrapper
         object Entity { get; }
         Document GetDocumentIfDirty();
         void Commit();
+        Document AsDocument();
```

Comment:
No need in this method.

We only need to convert an _entity_ into a Document, if that entity was modified or newly added. Simply because we only need to save modified Documents to storage, unchanged (and deleted) documents do not need to be saved. The GetDocumentIfDirty() method was specifically designed to underscore this.

If at some place of your code you found yourself wanting AsDocument() method, that rather signals a design flaw. And that's what we see in [TableDefinitionWrapper, 230](https://github.com/danieljurek/linq2dynamodb/blob/89651b8632591740c5ff5ec0cd96247186022a8e/Sources/Linq2DynamoDb.DataContext/TableDefinitionWrapper.cs#L230): no need to convert deleted entities into Documents, keys are enough for doing a delete.